### DISCUSSION TITLE
Use `prettyplease` for codegen formatting

### DISCUSSION THREAD

**Discussion by malobre** [rome/tools#1937] [Suggestions]
I'm currently working on #1932 where I need to format generated code, it is currently done by checking if `rustfmt` is installed and calling its binary: https://github.com/rome/tools/blob/7991631235ed2d7cd2d59b002cae75e78f82c80b/xtask/src/lib.rs#L52-L61

D. Tolnay just published a new crate called [`prettyplease`](https://github.com/dtolnay/prettyplease) made for this use-case (pretty-printing generated code).

According to `prettyplease` readme:
- it's faster than `rustfmt`
- it only depends on `syn` & `proc-macro2`, which we already use
- it can be ergonomically called, unlike `rustfmt`

Thoughts ?

**Comment by ematipico** [comment]
Oh, so this crate is a formatter for `syn`? I wouldn't mind using it for our code generation. Considering that `syn`, `quote` and now `prettyplease` are maintained by the same author, I think the integration between them might be better compared to `rustfmt`.

**Reply by malobre** [reply]
Yep, it exposes a single function `prettyplease::unparse` which takes a `syn::File`.

In my case I had a `TokenStream` from `proc-macro2`, which can be easily converted:
```rust
let pretty = prettyplease::unparse(&syn::parse2::<syn::File>(tokens)?);
```

**Comment by yassere** [comment]
I haven't looked closely at `prettyplease`, but I certainly like the idea of using a purpose-built library by `dtolnay` rather than shelling out to `rustfmt` for our codegen.
